miso-web/src/main/webapp/scripts/list_samplepurpose.js

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2012. The Genome Analysis Centre, Norwich, UK
+ * MISO project contacts: Robert Davey @ TGAC
+ * *********************************************************************
+ *
+ * This file is part of MISO.
+ *
+ * MISO is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MISO is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MISO.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * *********************************************************************
+ */
+
+ListTarget.samplepurpose = {
+  name : "Sample Purposes",
+  createUrl : function(config, projectId) {
+    throw "Must be provided statically";
+  },
+  createBulkActions : function(config, projectId) {
+    return config.isAdmin ? HotTarget.samplepurpose.bulkActions : [];
+  },
+  createStaticActions : function(config, projectId) {
+    return config.isAdmin ? [ {
+      "name" : "Add",
+      "handler" : function() {
+        
+        Utils.showDialog('Create Sample Purposes', 'Create', [ {
+          property : 'quantity',
+          type : 'int',
+          label : 'Quantity',
+          value : 1
+        } ], function(result) {
+          if (result.quantity < 1) {
+            alert("That's a peculiar number of samplepurposes to create.");
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this usage of alert(...). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)